### PR TITLE
Refactor format tests

### PR DESCRIPTION
**Pull request by phw**
# Summary

Cleaned up the formats tests and moved them into separate files. Also fixed some Vorbis tests and added additional tests for validating the read file info.

* This is a…
    * [ ] Bug fix
    * [ ] Feature addition
    * [x] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): [PICARD-XXX](https://tickets.metabrainz.org/browse/PICARD-XXX)

# Solution

# Action

### CODE REVIEW

**Comment by phw** [comment]
@zas So I decided to split up the cover art files as well as you suggested. That test file was a bit of a mess, I unified it a bit using the file handling we already have for other cases. This way we now also have all tests for a specific tag format in one file. Also some of the cover art tests where not specific to saving tags to files but where only testing the cover art classes. I moved those that were not redundant over to `test_coverart_image.py`.

I also added tests for utility functions defined in some of the format classes.

I am now finally happy with this for review.

**Comment by phw** [comment]
The last commit fixes [PICARD-1499](https://tickets.metabrainz.org/browse/PICARD-1499)

File: test/formats/test_asf.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        '~sample_rate': '44100',
+        '~bitrate': '64.0',
+    }
+
+
+class AsfUtilTest(PicardTestCase):
+    def test_pack_and_unpack_image(self):
+        mime = 'image/png'
+        image_data = create_fake_png(b'x')
+        type = 4
```
Comment (zas) [inline test/formats/test_asf.py]: I'd rather use another name, as `type` is a [builtin](https://docs.python.org/3/library/functions.html#type). Not critical though.

File: test/test_coverart_image.py [hunk 2]
```diff
@@ -84,6 +87,27 @@ def test_compare_with_multiple_types(self):
         self.assertEqual(image2, image3)
         self.assertNotEqual(image2, image4)
 
+    def test_set_data(self):
+        self.addCleanup(QtCore.QObject.tagger.run_cleanup)
+        imgdata = data=create_fake_png(b'a')
```
[the remark this answers is not in the document]
Comment (phw) [reply test/test_coverart_image.py]: Good catch. `data` was unused, that was a copy'n'paste leftover

File: test/formats/common.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+            return copy
+
+    class SimpleFormatsTestCase(BaseFileTestCase):
+
+        expected_info = None
+
+        @skipUnlessTestfile
+        def test_can_open_and_save(self):
+            metadata = Metadata()
+            metadata = save_and_load_metadata(self.filename, metadata)
```
[the remark this answers is not in the document]
Comment (phw) [reply test/formats/common.py]: There were only two occurences of calling `save_and_load_metadata()` with empty Metadata, so I rewrote them to a single line. I decided against making it default to empty metadata, as I think that somewhat hides what is happening and I think being explicit here is important in tests

File: test/formats/common.py [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+            new_metadata = save_and_load_metadata(self.filename, metadata)
+            self.assertNotIn('albumartist', new_metadata.keys())
+            self.assertNotIn('performer:drums', new_metadata.keys())
+            self.assertNotIn('totaltracks', new_metadata.keys())
+
+        @skipUnlessTestfile
+        def test_delete_complex_tags(self):
+            metadata = Metadata(self.tags)
+            original_metadata = save_and_load_metadata(self.filename, metadata)
+            metadata.delete('totaldiscs')
```
[the remark this answers is not in the document]
Comment (phw) [reply test/formats/common.py]: Yes, I used `del` now everywhere in the tests.

**Pull request merged**